### ISSUE TITLE
[BUG] Erroneous E_DISPATCHER_STATUS events

### ISSUE THREAD

**Issue by bcnewlin** [OpenSIPS/opensips#1816]
**OpenSIPS version you are running**

```
version: opensips 2.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 4e258b17f
main.c compiled on 20:10:26 Sep  3 2019 with gcc 7
```

**Describe the bug**

E_DISPATCHER_STATUS events are delivered out of order, so the last event raised does not always reflect the actual state of the dispatcher. This renders the events mostly useless. In addition, events are often raised when the state of the endpoint has not changed. Lastly, while reproducing this I found that ds_mark_dst does not work unless ds_select_dst/ds_select_domain is called with the "F" flag, which is not documented.

**To Reproduce**

You can use this
[OpenSIPS config file](https://github.com/OpenSIPS/opensips/files/3572352/opensips.cfg.txt)
 and
[SIPp scenario](https://github.com/OpenSIPS/opensips/files/3572354/uac.xml.txt) to reproduce reliably. Simply configure your dispatcher with an endpoint in group 1 that will not respond, like this:
```
# psql -c 'SELECT * FROM dispatcher;'
 id | setid |   destination    | socket | state | weight | attrs | description | version | instance_id | priority
----+-------+------------------+--------+-------+--------+-------+-------------+---------+-------------+----------
  1 |     1 | sip:0.0.0.0:8888 |        |     0 | 1      |       |             |         |             |        0
```

You will see that each call to ds_mark_dst raises a E_DISPATCHER_STATUS event, even if the status of the endpoint is not changed due to not hitting the threshold. You will then see that the last call to ds_mark_dst to hit the threshold causes 2 E_DISPATCHER_STATUS events; one inactive, which is correct, followed by one stating the endpoint is active, which is not correct. With the sample configuration which is set to probe_mode 0, the dispatcher will now begin sending OPTIONS pings, which proves the endpoint is in PROBING state, despite the last event indicating active.

**Expected behavior**

E_DISPATCHER_STATUS events should only be raised when the status of the endpoint changes, as is indicated in the documentation. The last event received for an endpoint should indicate the current status.

**Relevant System Logs**

Attached are the logs from running the above sample scenario/configuration for 2 calls.
[dispatcher.txt](https://github.com/OpenSIPS/opensips/files/3572374/dispatcher.txt)

**OS/environment information**
 - Operating System:  Amazon Linux 2 in Docker
 - OpenSIPS installation:  git
 - other relevant information:

**Additional context**

**Comment by bogdan-iancu** [comment]
Funny, this issue was already fixed in 3.0, but not backported to 2.4

**Comment by bcnewlin** [comment]
@bogdan-iancu Thanks!
